bin/desi_download_dr1

```diff
@@ -0,0 +1,294 @@
+        return False
+
+def get_tile_date(remote_base_url, tileid, auth=None):
+    """Get the most recent date directory for a tile"""
+    tile_url = f"{remote_base_url}spectro/redux/iron/tiles/cumulative/{tileid}/"
+    contents = list_directory(tile_url, auth)
+    
+    if contents is None:
+        if auth is None:
+            try:
+                user, pwd = get_desi_login_password()
+                return get_tile_date(remote_base_url, tileid, (user, pwd))
+            except Exception:
+                return None
+        return None
+    
+    # Filter for date-like directories (8 digits) and get the latest
+    date_dirs = [os.path.basename(d.rstrip('/')) for d in contents if d.rstrip('/').split('/')[-1].isdigit() and len(d.rstrip('/').split('/')[-1]) == 8]
+    return max(date_dirs) if date_dirs else None
+
+def analyze_tiles(redrock_file, max_tiles=1):
+    """
+    Analyze tileIDs from redrock file and return top N tiles by target count
+    
+    Parameters:
+        redrock_file (str): Path to redrock FITS file
+        max_tiles (int): Maximum number of tiles to return (default: 3)
+    """
+    try:
+        r = Table.read(redrock_file, hdu=3)
+        unique_values, counts = np.unique(r['TILEID'], return_counts=True)
+        
+        # Sort tiles by count in descending order
+        sorted_indices = np.argsort(counts)[::-1]
+        sorted_tileids = unique_values[sorted_indices]
+        sorted_counts = counts[sorted_indices]
+        
+        print("\nTile analysis (sorted by target count):")
+        for tileid, count in zip(sorted_tileids, sorted_counts):
+            print(f"TILEID {tileid}: {count} targets")
+        
+        # Return top N tiles
+        top_tiles = sorted_tileids[:max_tiles]
+        print(f"\nSelecting top {max_tiles} tiles: {', '.join(map(str, top_tiles))}")
+        return top_tiles
+    except Exception as e:
+        print(f"Error analyzing redrock file: {str(e)}")
+        return None
+
+def main():
+    parser = argparse.ArgumentParser(description='Download DESI DR1 data for targets near a position')
+    parser.add_argument('--ra', type=float, default=150.11916667, help='Center right ascension in degrees (default: 150.11916667)')
+    parser.add_argument('--dec', type=float, default=2.20583333, help='Center declination in degrees (default: 2.20583333)')
+    parser.add_argument('--radius', type=float, default=0.5, help='Search radius in degrees (default: 0.5)')
+    parser.add_argument('--base-dir', default='./tiny_dr1', help='Base directory for downloads')
+    args = parser.parse_args()
+    
+    remote_base_url = 'https://data.desi.lbl.gov/public/dr1/'
+    local_base_path = args.base_dir
+    
+    print(f"Starting downloads from {remote_base_url}")
+    print(f"Files will be saved to {local_base_path}")
+    print(f"\nSearching for targets around RA={args.ra}, Dec={args.dec} with radius={args.radius} degrees")
+    
+    # Download redshift catalog first
```

Comment:
For the purposes of making a tutorial subset, let's also download the `{specprod}/tiles-{specprod}.csv` and `{specprod}/exposures-{specprod}.csv` files (or the .fits versions).

Reply:
Done